build/yamls/antrea-node-init.yml

```diff
@@ -0,0 +1,42 @@
+kind: DaemonSet
+apiVersion: extensions/v1beta1
+metadata:
+  name: antrea-node-init
+  namespace: kube-system
+  labels:
+    app: antrea-node-init
+spec:
+  template:
+    metadata:
+      labels:
+        app: antrea-node-init
+    spec:
+      hostPID: true
+      hostNetwork: true
+      containers:
+        - name: node-init
+          image: gcr.io/google-containers/startup-script:v1
```

Comment:
I checked how this image runs, I guess it uses a loop as a workaround because there's no long running jobs. However, since we have long running container in antrea-agent, do you think we can just make this another init container of that daemonset? or restarting kubelet make combining them not appropriate?

Reply:
I think an init container in agent DS is good in that it can make sure Node is initialized before agent starts. But it also sounds a little bad to me we add the hack like kubelet restart to agent DS. Maybe it is just my personal thinking that does not make sense. @tnqn and @reachjainrahul : please share your opinions.

Reply:
Its safe to not restart Kubelet while pods are launched.

Reply:
@reachjainrahul what did you mean? Not restart kubelet or restart kubelet? My point is whether they can be combined in the same daemonset as antrea-agent to save one yaml and one step to install antrea.
@jianjuns I think the problem is whether there's a better approach to enable CNI for kubelet. Rahul and I had a look at how other CNIs do, this seems the only approach. Is your point the hack itself or adding it to antrea-agent DS?

Reply:
@tnqn Couple of things
- This is temporary. In our final integrated solution, this step will go away
- Secondly, yes, we can update our init container but it has to use different image (as our ubuntu image doesnt have systemD) and kubelet restart requires it.
- Lastly I feel its not safe to restart kubelet while antrea pods are deployed. In this case, once the node-init pod is deployed, it then restarts kubelet

Reply:
I now recall the different image problem. Kindof agree a separate DS is better.

Reply:
I think init containers and normal containers can use different images, but I agree with the 1st and 3rd points.